algorithms/unit_tests/TestStdAlgorithmsTeamFind.cpp

```diff
@@ -0,0 +1,190 @@
+          if (it != KE::cend(myRowViewFrom)) {
+            m_destView(myRowIndex) = *it;
+          }
+
+          m_distancesView(myRowIndex) =
+              KE::distance(KE::cbegin(myRowViewFrom), it);
+        });
+
+        break;
+      }
+
+      case 1: {
+        auto it = KE::find(member, myRowViewFrom, m_value);
+
+        Kokkos::single(Kokkos::PerTeam(member), [=]() {
+          if (it != KE::end(myRowViewFrom)) {
+            m_destView(myRowIndex) = *it;
+          }
+
+          m_distancesView(myRowIndex) =
+              KE::distance(KE::begin(myRowViewFrom), it);
+        });
+
+        break;
+      }
+    }
+  }
+};
+
+template <class LayoutTag, class ValueType>
+void test_A(std::size_t numTeams, std::size_t numCols, int apiId) {
+  /* description:
+     use a rank-2 view randomly filled with values,
+     and run a team-level find
+   */
+
+  // -----------------------------------------------
+  // prepare data
+  // -----------------------------------------------
+  // create a view in the memory space associated with default exespace
+  // with as many rows as the number of teams and fill it with random
+  // values from an arbitrary range.
+  auto [sourceView, sourceViewBeforeOp_h] = create_random_view_and_host_clone(
+      LayoutTag{}, numTeams, numCols,
+      Kokkos::pair{ValueType(5), ValueType(523)}, "sourceView");
+
+  // -----------------------------------------------
+  // launch kokkos kernel
+  // -----------------------------------------------
+  using space_t = Kokkos::DefaultExecutionSpace;
+  Kokkos::TeamPolicy<space_t> policy(numTeams, Kokkos::AUTO());
+
+  // to verify that things work, each team stores the result of its find call,
+  // and then we check that these match what we expect
+  Kokkos::View<ValueType*> destView("destView", numTeams);
+
+  // find returns an iterator so to verify that it is correct
+  // each team stores the distance of the returned iterator from the
+  // beginning of the interval that team operates on and then we check
+  // that these distances match the std result
+  Kokkos::View<std::size_t*> distancesView("distancesView", numTeams);
+
+  // use CTAD for functor
+  const ValueType value{sourceViewBeforeOp_h(numTeams / 2, numCols / 2)};
+  TestFunctorA fnc(sourceView, destView, distancesView, value, apiId);
+  Kokkos::parallel_for(policy, fnc);
+
+  // -----------------------------------------------
+  // run cpp-std kernel and check
+  // -----------------------------------------------
+  auto destView_h      = create_host_space_copy(destView);
+  auto distancesView_h = create_host_space_copy(distancesView);
+  Kokkos::View<ValueType**, Kokkos::HostSpace> stdDestView("stdDestView",
+                                                           numTeams, numCols);
+  for (std::size_t i = 0; i < sourceView.extent(0); ++i) {
+    auto rowFrom = Kokkos::subview(sourceViewBeforeOp_h, i, Kokkos::ALL());
+    auto rowDest = Kokkos::subview(stdDestView, i, Kokkos::ALL());
```

Comment:
not used anywhere

Reply:
My reasoning here was similar to adjacent_find. Removed.